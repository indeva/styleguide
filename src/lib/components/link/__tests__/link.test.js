import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Link from '..';

describe('Link', () => {
  describe('with url', () => {
    it('should render link properly', () => {
      const component = renderer.create(<Link url="/test">link with href</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with target', () => {
    it('should render link properly', () => {
      const component = renderer.create(<Link url="/test" target="_blank">link</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with custom class', () => {
    it('should render link properly', () => {
      const component = renderer.create(<Link url="/test" className="my classes">my link</Link>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('behavior tests', () => {
    describe('when is clicked', () => {
      it('calls the callback function', () => {
        const onClickCallback = jest.fn();
        const wrapper = mount(
          <Link onClick={onClickCallback} url="/test" className="links">my link</Link>,
        );

        wrapper.find('.link').simulate('click');
        expect(onClickCallback).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('with modifier', () => {
    describe('primary', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" primary>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('secondary', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" secondary>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('tertiary', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" tertiary>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('small', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" small>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('medium', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" medium>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('large', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" large>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('block', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" block>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('compose by primary and large and block', () => {
      it('should render link properly', () => {
        const component = renderer.create(<Link url="/test" primary large block>Test</Link>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
