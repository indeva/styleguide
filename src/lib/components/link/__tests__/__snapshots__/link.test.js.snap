// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with custom class should render link properly 1`] = `
<a
  className="link my classes"
  href="/test"
  target="_self"
>
  my link
</a>
`;

exports[`Link with modifier block should render link properly 1`] = `
<a
  className="link link--block"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier compose by primary and large and block should render link properly 1`] = `
<a
  className="link link--primary link--large link--block"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier large should render link properly 1`] = `
<a
  className="link link--large"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier medium should render link properly 1`] = `
<a
  className="link link--medium"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier primary should render link properly 1`] = `
<a
  className="link link--primary"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier secondary should render link properly 1`] = `
<a
  className="link link--secondary"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier small should render link properly 1`] = `
<a
  className="link link--small"
  href="/test"
  target="_self"
>
  Test
</a>
`;

exports[`Link with modifier tertiary should render link properly 1`] = `
<a
  className="link link--tertiary"
  target="_self"
>
  Test
</a>
`;

exports[`Link with target should render link properly 1`] = `
<a
  className="link"
  href="/test"
  target="_blank"
>
  link
</a>
`;

exports[`Link with url should render link properly 1`] = `
<a
  className="link"
  href="/test"
  target="_self"
>
  link with href
</a>
`;
